src/modules/deno.rs

```diff
@@ -0,0 +1,108 @@
+use super::{Context, Module, RootModuleConfig};
+
+use crate::configs::deno::DenoConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with the current Deno version
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("deno");
+    let config = DenoConfig::try_load(module.config);
+    let is_deno_project = context
+        .try_begin_scan()?
+        .set_files(&config.detect_files)
+        .is_match();
+
+    if !is_deno_project {
+        return None;
+    }
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_meta(|var, _| match var {
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => context
+                    .exec_cmd("deno", &["-V"])
+                    .and_then(|output| parse_deno_version(output.stdout.trim()))
+                    .map(Ok),
+                _ => None,
+            })
+            .parse(None)
+    });
+
+    module.set_segments(match parsed {
+        Ok(segments) => segments,
+        Err(error) => {
+            log::warn!("Error in module `deno`:\n{}", error);
+            return None;
+        }
+    });
+
+    Some(module)
+}
+
+fn parse_deno_version(deno_version: &str) -> Option<String> {
+    let version = deno_version
+        // split into ["deno 1.8.3", ""]
+        .split('\n')
+        // get first line
+        .next()
+        // unwrap
+        .unwrap()
```

Comment:
I'd prefer this (no `unwrap`, `split('\n')` -> `lines()`):
```suggestion
        // split into ["deno 1.8.3", ""]
        .lines()
        // get first line
        .next()?
```

But it even works without these lines for me:
```suggestion
```

Reply:
For me it didn't work without the long lines of code, but, if it works for you then I suppose I can try again.

Okay, it worked without it. Wasn't working yesterday so must of been a fluke.